src/game/behaviors/fish.inc.c

```diff
@@ -1,163 +1,233 @@
+/**
+ * Set certain settings such as animations, colour, and spawn quanitity.
+ * Chosen settings dependant on parameter o->oBehParams2ndByte
+ */
+void fish_group_act_spawn(void) {
     s32 i;
-    s32 sp38;
+    s32 schoolQuantity;
     s16 model;
-    f32 sp30;
-    struct Animation **sp2C;
-    struct Object *sp28;
+    f32 minDistToMario;
+    struct Animation **fishAnimation;
+    struct Object *fishObject;
     switch (o->oBehParams2ndByte) {
         case 0:
-            model = MODEL_FISH;            sp38 = 20;            sp30 = 1500.0f;            sp2C = blue_fish_seg3_anims_0301C2B0;
+            model = MODEL_FISH;            schoolQuantity = 20;            minDistToMario = 1500.0f;            fishAnimation = blue_fish_seg3_anims_0301C2B0;
             break;
         case 1:
-            model = MODEL_FISH;            sp38 = 5;            sp30 = 1500.0f;            sp2C = blue_fish_seg3_anims_0301C2B0;
+            model = MODEL_FISH;            schoolQuantity = 5;            minDistToMario = 1500.0f;            fishAnimation = blue_fish_seg3_anims_0301C2B0;
             break;
         case 2:
-            model = MODEL_CYAN_FISH;            sp38 = 20;            sp30 = 1500.0f;            sp2C = cyan_fish_seg6_anims_0600E264;
+            model = MODEL_CYAN_FISH;            schoolQuantity = 20;            minDistToMario = 1500.0f;            fishAnimation = cyan_fish_seg6_anims_0600E264;
             break;
         case 3:
-            model = MODEL_CYAN_FISH;            sp38 = 5;            sp30 = 1500.0f;            sp2C = cyan_fish_seg6_anims_0600E264;
+            model = MODEL_CYAN_FISH;            schoolQuantity = 5;            minDistToMario = 1500.0f;            fishAnimation = cyan_fish_seg6_anims_0600E264;
             break;
     }
-    if (o->oDistanceToMario < sp30 || gCurrLevelNum == LEVEL_SA) {
-        for (i = 0; i < sp38; i++) {
-            sp28 = spawn_object(o, model, bhvFishGroup2);
-            sp28->oBehParams2ndByte = o->oBehParams2ndByte;
-            func_8029EE20(sp28, sp2C, 0);
-            translate_object_xyz_random(sp28, 700.0f);
+    // If Distance to Mario is lower than 1500 or the current level is Secret Aquarium,
+    // then spawn the schoolQuantity of fish. Fish moves at random.
+    if (o->oDistanceToMario < minDistToMario || gCurrLevelNum == LEVEL_SA) {
+        for (i = 0; i < schoolQuantity; i++) {
+            fishObject = spawn_object(o, model, bhvFishGroup2);
+            fishObject->oBehParams2ndByte = o->oBehParams2ndByte;
+            func_8029EE20(fishObject, fishAnimation, 0);
+            translate_object_xyz_random(fishObject, 700.0f);
         }
         o->oAction = 1;
     }
 }
 
-void ActionFish1(void) {
-    if (gCurrLevelNum != LEVEL_SA)
-        if (gMarioObject->oPosY - o->oPosY > 2000.0f)
+// If the current level is not Secret Aquarium and the distance from
+// mario minus o->oPosY is greater than 2000.0f then remove fish.
+void bhv_fish_action_1(void) {
```

Comment:
This is not a sufficient name. I realize that neither is the old one, but it would be worth taking care of it here. Additionally, this comment should be a block style comment.

Also, bhv_ is reserved for things called in behavior_data. This should just have the formatting fish_act_XXX